File: frontend/src/services/cartService.js
// src/services/cartService.js
import api from './api';

export const getCart = () => api.get('/cart');
export const addToCart = (productId, quantity) => api.post('/cart/add', { productId, quantity });
export const updateCartItem = (productId, quantity) => api.put('/cart/update', { productId, quantity });
export const removeFromCart = (productId) => api.delete(`/cart/${productId}`);
export const clearCart = () => api.delete('/cart/clear');

File: src/main/java/com/ebizzness/ecommerce/service/CartService.java
package com.ebizzness.ecommerce.service;

import com.ebizzness.ecommerce.dto.request.AddToCartRequest;
import com.ebizzness.ecommerce.dto.request.UpdateCartItemRequest;
import com.ebizzness.ecommerce.dto.response.CartResponse;
import org.springframework.stereotype.Service;

@Service
public interface CartService {
    CartResponse addToCart(AddToCartRequest request, String authorizationHeader);
    CartResponse removeFromCart(Long productId, String authorizationHeader);
    CartResponse updateCartItem(UpdateCartItemRequest request, String authorizationHeader);
    CartResponse getCart(String authorizationHeader);
    void clearCart(String authorizationHeader);
}